### PR TITLE
Right drawer to edit records

### PR DESCRIPTION
**Pull request by FelixMalfait**
This PR introduces a new side panel to edit records and the ability to minimize the side panel.

The goal is leverage this sidepanel to be able to create records while being in another show page.

I'm opening the PR for feedback since it involved refactoring and therefore already touches a lot of files, even though it was quick to implement.

Screenshot 2024-05-23 at 17 41 37

### CODE REVIEW

**Comment by charlesBochet** [comment]
Overall the direction LGTM, I'll have a closer look tomorrow

**Comment by lucasbordeau** [comment]
Bug when we switch page, an animation plays but it shouldn't do anything when the drawer is folded.


https://github.com/twentyhq/twenty/assets/26528466/2f568f83-d524-463c-a516-20408e20d6f5

**Comment by lucasbordeau** [comment]
Should it be clickable ? Seems like it should be something display only.
image

Also it should have a padding-left of 8px to match the 16px gap of top and bottom.

File: packages/twenty-front/src/modules/activities/calendar/right-drawer/hooks/useOpenCalendarEventRightDrawer.ts [hunk 1]
```diff
@@ -9,9 +9,7 @@ import { useSetHotkeyScope } from '@/ui/utilities/hotkey/hooks/useSetHotkeyScope
 export const useOpenCalendarEventRightDrawer = () => {
   const { openRightDrawer } = useRightDrawer();
   const setHotkeyScope = useSetHotkeyScope();
-  const setViewableCalendarEventId = useSetRecoilState(
-    viewableCalendarEventIdState,
-  );
+  const setViewableCalendarEventId = useSetRecoilState(viewableRecordIdState);
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/src/modules/activities/calendar/right-drawer/hooks/useOpenCalendarEventRightDrawer.ts:12]: Yes

File: packages/twenty-front/src/modules/activities/comment/__stories__/Comment.stories.tsx [hunk 2]
```diff
@@ -3,15 +3,15 @@ import { Meta, StoryObj } from '@storybook/react';
...
+import { viewableRecordIdState } from '@/object-record/record-right-drawer/states/viewableRecordIdState';
 
 import { ActivityActionBar } from '../../right-drawer/components/ActivityActionBar';
 import { Comment } from '../Comment';
 
 import { mockComment, mockCommentWithLongValues } from './mock-comment';
 
 const CommentSetterEffect = () => {
-  const setViewableActivity = useSetRecoilState(viewableActivityIdState);
+  const setViewableActivity = useSetRecoilState(viewableRecordIdState);
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/activities/comment/__stories__/Comment.stories.tsx:14]: Rename here too

File: packages/twenty-front/src/modules/activities/comment/__stories__/CommentHeader.stories.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { CommentHeader } from '../CommentHeader';
 import { mockComment, mockCommentWithLongValues } from './mock-comment';
 
 const CommentHeaderSetterEffect = () => {
-  const setViewableActivity = useSetRecoilState(viewableActivityIdState);
+  const setViewableActivity = useSetRecoilState(viewableRecordIdState);
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/activities/comment/__stories__/CommentHeader.stories.tsx:16]: Rename here too

File: packages/twenty-front/src/modules/activities/emails/hooks/useEmailThread.ts [hunk 4]
```diff
@@ -17,19 +17,19 @@ export const useEmailThread = () => {
           .getValue();
 
         const viewableEmailThreadId = snapshot
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/activities/emails/hooks/useEmailThread.ts:19]: viewableRecordId

File: packages/twenty-front/src/modules/activities/hooks/useOpenCreateActivityDrawer.ts [hunk 5]
```diff
@@ -26,7 +26,7 @@ export const useOpenCreateActivityDrawer = () => {
   const setActivityTargetableEntityArray = useSetRecoilState(
     activityTargetableEntityArrayState,
   );
-  const setViewableActivityId = useSetRecoilState(viewableActivityIdState);
+  const setViewableActivityId = useSetRecoilState(viewableRecordIdState);
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/activities/hooks/useOpenCreateActivityDrawer.ts:29]: setViewableRecordId

File: packages/twenty-front/src/modules/object-record/record-show/components/RecordShowContainer.tsx [hunk 6]
```diff
@@ -129,114 +132,118 @@ export const RecordShowContainer = ({
...
-                    </FieldContext.Provider>
-                  ))
-                )}
-              </PropertyBox>
-              <RecordDetailDuplicatesSection
-                objectRecordId={objectRecordId}
-                objectNameSingular={objectNameSingular}
-              />
-              {relationFieldMetadataItems?.map((fieldMetadataItem, index) => (
+  const summary = (
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-show/components/RecordShowContainer.tsx:138]: Code smell, using a function is a sign that the component should be broken down in multiple smaller components

File: packages/twenty-front/src/modules/object-record/relation-picker/components/RelationPicker.tsx [hunk 7]
```diff
@@ -39,11 +42,32 @@ export const RelationPicker = ({
...
+        fieldDefinition.metadata.relationObjectMetadataNameSingular,
+    });
+
+  const relationFieldMetadataItem = relationObjectMetadataItem.fields.find(
+    ({ id }) => id === fieldDefinition.metadata.relationFieldMetadataId,
+  );
+
+  const { entityId } = useContext(FieldContext);
+
+  const { handleOnCreate } = useAddNewRecordAndOpenRightDrawer({
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/relation-picker/components/RelationPicker.tsx:57]: handleOnCreate should be the responsibility of a component, useAddNewRecordAndOpenRightDrawer should just expose createNewRecordAndOpenRightDrawer.

File: packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx [hunk 8]
```diff
@@ -71,6 +71,18 @@ export const SingleEntitySelectMenuItemsWithSearch = ({
     objectNameSingular: relationObjectNameSingular,
   });
 
+  const onCreateWithInput = isDefined(onCreate)
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx:73]: handleCreate = () => {
  if(!isDefined(onCreate)) {
    return
  } 

  if (onCreate.length > 0) {
          (onCreate as (searchInput?: string) => void)(
            relationPickerSearchFilter,
          );
        } else {
          (onCreate as () => void)();
        }
}

File: packages/twenty-front/src/modules/object-record/relation-picker/hooks/useAddNewRecordAndOpenRightDrawer.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+                  newRecordId,
+              },
+            });
+          }
+
+          setViewableRecordId(newRecordId);
+          setViewableRecordNameSingular(relationObjectMetadataNameSingular);
+          openRightDrawer(RightDrawerPages.ViewRecord);
+        }
+      : undefined;
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/relation-picker/hooks/useAddNewRecordAndOpenRightDrawer.ts:101]: Hard to read with a big ternary like that, prefer early return at the beginning of the function.

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawerTopBarExpandButton.tsx [hunk 10]
```diff
@@ -1,20 +1,17 @@
...
 export const RightDrawerTopBarExpandButton = () => {
-  const [isRightDrawerExpanded, setIsRightDrawerExpanded] = useRecoilState(
-    isRightDrawerExpandedState,
-  );
+  const { isRightDrawerExpanded, downsizeRightDrawer, expandRightDrawer } =
+    useRightDrawer();
 
   const handleButtonClick = () => {
-    setIsRightDrawerExpanded(!isRightDrawerExpanded);
+    isRightDrawerExpanded ? downsizeRightDrawer() : expandRightDrawer();
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawerTopBarExpandButton.tsx:14]: If / else

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawerTopBarMinimizeButton.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+import { LightIconButton } from '@/ui/input/button/components/LightIconButton';
+import { useRightDrawer } from '@/ui/layout/right-drawer/hooks/useRightDrawer';
+
+export const RightDrawerTopBarMinimizeButton = () => {
+  const { isRightDrawerMinimized, minimizeRightDrawer, maximizeRightDrawer } =
+    useRightDrawer();
+
+  const handleButtonClick = () => {
+    isRightDrawerMinimized ? maximizeRightDrawer() : minimizeRightDrawer();
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawerTopBarMinimizeButton.tsx:11]: if / else

File: packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx (hunk 8 above)
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx:73]: That would make it look like handleCreate is defined (with () => {return}). But I'll try to simplify!

**Comment by FelixMalfait** [comment]
cc @pombredanne fyi, sorry it took so long! We finally enabled the ability to create records from a side panel, without leaving the main page. Looking forward to your feedback

Fixes #4901

Will be deployed in the next release in 2 weeks

**Pull request merged**